pkg/gce-pd-csi-driver/controller.go

```diff
@@ -78,29 +79,52 @@ func (gceCS *GCEControllerServer) CreateVolume(ctx context.Context, req *csi.Cre
 	// Apply Parameters (case-insensitive). We leave validation of
 	// the values to the cloud provider.
 	diskType := "pd-standard"
-	configuredZone := gceCS.CloudProvider.GetZone()
+	var configuredZone string
+
 	for k, v := range req.GetParameters() {
 		if k == "csiProvisionerSecretName" || k == "csiProvisionerSecretNamespace" {
 			// These are hardcoded secrets keys required to function but not needed by GCE PD
 			continue
 		}
 		switch strings.ToLower(k) {
-		case "type":
+		case common.ParameterKeyType:
 			glog.Infof("Setting type: %v", v)
 			diskType = v
-		case "zone":
+		case common.ParameterKeyZone:
 			configuredZone = v
 		default:
 			return nil, status.Error(codes.InvalidArgument, fmt.Sprintf("invalid option %q", k))
 		}
 	}
 
+	// Topology overwrites Storage Class Params if specified
+	// TODO: Support preferred toplogies.
+	if req.GetAccessibilityRequirements() != nil {
+		if len(configuredZone) != 0 {
+			return nil, status.Error(codes.InvalidArgument, fmt.Sprintf("CreateVolume only one of parameter zone or topology zone may be specified"))
+		}
+		configuredZone, err = pickTopology(req.GetAccessibilityRequirements())
+		if err != nil {
+			return nil, status.Error(codes.InvalidArgument, fmt.Sprintf("CreateVolume failed to pick topology: %v", err))
+		}
+	}
+
+	if len(configuredZone) == 0 {
+		// Default to zone that the driver is in
+		configuredZone = gceCS.CloudProvider.GetZone()
```

Comment:
The zone-picking strategy should be consistent, regardless of whether zones are specified through parameters or AccessibilityRequirement

Reply:
what do you mean by this? consistent with what?

Reply:
I think what @verult means is that once you enable the topology capability, then you should always get an accessibility requirement.  So if you run into a case where there is no zone specified, then you should error.

Reply:
I believe accessibility requirements can be unset:
```
// If this field is not specified and the SP has the
// ACCESSIBILITY_CONSTRAINTS plugin capability, the SP MAY choose
// where the provisioned volume is accessible from.
```

The current logic is:
1) Only one of `Parameters` or `AccessibilityRequirements` may be set. Error if both are set.
2) If neither are set (valid according to CSI Spec) default to zone that driver is in

This logic makes sense to me according to the Spec. Please let me know if you think it should be changed and we can discuss